app/file.go

```diff
@@ -623,32 +623,46 @@ func (a *App) UploadFileX(channelId, name string, input io.Reader,
 		}
 	}
 
-	aerr = t.readAll()
+	written, aerr := t.writeFile(io.MultiReader(t.buf, t.limitedInput), t.fileinfo.Path)
 	if aerr != nil {
-		return t.fileinfo, aerr
+		return nil, aerr
 	}
 
-	aerr = t.runPlugins()
+	if written > t.maxFileSize {
+		a.RemoveFile(t.fileinfo.Path)
```

Comment:
Do we want to log the error returned from here?

Reply:
Yes, that can't hurt.